python/pyspark/sql/types.py

```diff
@@ -2197,8 +2197,8 @@ def verify_nullability(obj: Any) -> bool:
                 return True
             else:
                 raise PySparkValueError(
-                    error_class="CANNOT_BE_NONE",
-                    message_parameters={"arg_name": "obj"},
+                    error_class="NULLABILITY_CHECK_FAILED",
+                    message_parameters={},
```

Comment:
@itholic actually I think we have a regression here. https://github.com/apache/spark/commit/04816474bfcc05c7d90f7b7e8d35184d95c78cbd#diff-bb02965370d515ec7add7967bc949c18c88ffc7346cceaebefb84907dfdce903L1933

Reply:
We should call `new_msg` to make the error message properly shown.

Reply:
https://github.com/apache/spark/pull/40975 too.

Reply:
See https://github.com/apache/spark/pull/18521. We should have the same error messages as https://gist.github.com/HyukjinKwon/7a494e4557fe32a652ce1236e504a395

Reply:
Oh.. it sounds like `types.py` has its own error message generation logic - called `new_msg` - and some error messages are should be generated use it instead of using error class defined in `error_classes.py` ?

Reply:
So, maybe we should like

Before
```python
            raise PySparkTypeError(
                error_class="CANNOT_ACCEPT_OBJECT_IN_TYPE",
                message_parameters={
                    "data_type": str(dataType),
                    "obj_name": str(obj),
                    "obj_type": type(obj).__name__,
                },
```

After
```python
            raise PySparkTypeError(
                message=new_msg("%s can not accept object %r in type %s" % (dataType, obj, type(obj))),
            )
```
?

Reply:
Yeah new_msg reclusively track the column name

Reply:
Oh,, sorry I didn't realize that. Let me fix them within this PR and will update the PR title & description accordingly.

Thanks for the context!